src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/custom.py

```diff
@@ -0,0 +1,133 @@
+#---------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+#---------------------------------------------------------------------------------------------
+
+from azure.cli.core.commands import cli_command
+
+from azure.cli.command_modules.acr.containerregistry.models import (
+    RegistryParameters,
+    RegistryProperties,
+    StorageAccountProperties
+)
+
+from ._factory import (
+    get_registry_service_client,
+    get_storage_end_point_suffix
+)
+
+from ._arm_utils import (
+    arm_get_registries_in_subscription,
+    arm_get_registries_in_resource_group,
+    arm_get_registry_by_name,
+    arm_deploy_template
+)
+
+from ._utils import (
+    get_registry_by_name,
+    get_resource_group_by_registry,
+    validate_registry_name
+)
+
+from ._format import output_format
+
+import azure.cli.core._logging as _logging
+logger = _logging.get_az_logger(__name__)
+
+def acr_list(resource_group=None):
+    '''Returns the list of container registries.
+    :param str resource_group: The name of resource group
+    '''
+    if resource_group:
+        return arm_get_registries_in_resource_group(resource_group)
+    else:
+        return arm_get_registries_in_subscription()
+
+def acr_create(resource_group, registry_name, location, storage_account_name=None, #pylint: disable=too-many-arguments
+               storage_account_key=None, deployment_name="Microsoft.ContainerRegistry"):
+    '''Returns the created container registry.
+    :param str resource_group: The name of resource group
+    :param str registry_name: The name of container registry
+    :param str location: The name of location
+    :param str storage_account_name: The name of storage account
+    :param str storage_account_key: The key of storage account
+    :param str deployment_name: The name of deployment
+    '''
+    validate_registry_name(registry_name)
```

Comment:
If you register the parameter registry_name in your _params.py file, you can then remove the calls to `validate_registry_name()` elsewhere in this file.